### PR TITLE
handle cargo features flags

### PR DESCRIPTION
**Pull request by gibix**
Not sure about the API, if sounds good I can write an example before merging.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -247,6 +270,13 @@ pub fn metadata_deps(manifest_path: Option<&Path>, deps: bool) -> Result<Metadat
         cmd.arg("--no-deps");
     }
 
+    match feat {
+        CargoOpt::AllFeatures => cmd.arg("--features-all"),
+        CargoOpt::NoDefaultFeatures => cmd.arg("--no-default-features"),
+        CargoOpt::SomeFeatures(ftrs) => cmd.arg(format!("--fatures {:?}", ftrs)),
```
Comment (oli-obk) [inline src/lib.rs]: typo: `fatures`

File: src/lib.rs [hunk 2]
```diff
@@ -247,6 +270,13 @@ pub fn metadata_deps(manifest_path: Option<&Path>, deps: bool) -> Result<Metadat
...
+        CargoOpt::DefaultFeatures => cmd.arg(""),
```
Comment (oli-obk) [inline src/lib.rs]: This arm should be a nop `{}` arm instead of adding an empty argument

Comment (gibix) [reply src/lib.rs]: is not possible to give a nop arg because cmd.arg returns a Command. Suggestions?

Comment (oli-obk) [reply src/lib.rs]: Just make the last arm `cmd` instead of `cmd.arg("")`, that should do it.

**Comment by gibix** [comment]
changed to Option<CargoOpt>`

File: src/lib.rs [hunk 3]
```diff
@@ -247,6 +268,14 @@ pub fn metadata_deps(manifest_path: Option<&Path>, deps: bool) -> Result<Metadat
         cmd.arg("--no-deps");
     }
 
+    if features.is_some() {
```
Comment (oli-obk) [inline src/lib.rs]: `if let Some(features) = features` saves you the unwrap

Comment (gibix) [reply src/lib.rs]: done, thanks for the feedback, is a rust-beginner contribution ;)

**Comment by oli-obk** [comment]
>  I can write an example before merging.

Code looks good to me! Can you write this example so we have some form of testing for this feature?

**Comment by gibix** [comment]
ping

**Comment by oli-obk** [comment]
0.5.6 has been published with this feature

**Comment by gibix** [comment]
thanks

**Pull request merged**